### PR TITLE
AWS::AutoScaling::LaunchConfiguration.MetadataOptions

### LINKED ISSUE #1839: Add MetadataOptions support for LaunchConfiguration
The [MetadataOptions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-launchconfig.html#cfn-autoscaling-launchconfig-metadataoptions) are currently unsupported for [LaunchConfiguration](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-launchconfig.html). This is an issue because it prevents the user from [configuring IMDS version](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html) to comply with [security best practices](https://docs.aws.amazon.com/eks/latest/userguide/best-practices-security.html )

- A similar case was found with LaunchTemplateData: [https://github.com/cloudtools/troposphere/issues/1590](https://github.com/cloudtools/troposphere/issues/1590)
- #1590 was solved in #1582 (Similar solution to this issue)

### PR DESCRIPTION
**Pull request by sorojara**
This fixes #1839

### CODE REVIEW

File: troposphere/autoscaling.py [hunk 1]
```diff
@@ -181,6 +189,7 @@ class AutoScalingGroup(AWSObject):
         'LoadBalancerNames': (list, False),
         'MaxInstanceLifetime': (integer, False),
         'MaxSize': (integer, True),
+        'MetadataOptions': (MetadataOptions, False),
```
Comment (PatMyron) [inline troposphere/autoscaling.py:192]: [`AWS::AutoScaling::AutoScalingGroup`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-group.html) doesn't have this property
```suggestion
```

Comment (sorojara) [reply troposphere/autoscaling.py]: fixed

**Review by PatMyron — APPROVED** [review]

**Comment by charleshepner** [comment]
What timeframe are we looking at for this PR to be merged and released?  We've been watching this PR for a few weeks and are just curious for our planning.

**Comment by markpeek** [comment]
Thanks!

**Pull request merged**